java/parser/src/main/java/gov/lbl/parser/domain/Declaration.java

```diff
@@ -20,32 +22,27 @@ public Declaration(String ident,
     	this.name = ident;
     	this.array_subscripts = (array_subscripts == null ? null : array_subscripts);
     	if (modification != null) {
-    		if (modification.charAt(0) == '(') {
-    			Pattern pattern1 = Pattern.compile("\\)=");
-    			Pattern pattern2 = Pattern.compile("\\) =");
-    			Pattern pattern3 = Pattern.compile("\\)");
-    			Matcher matcher1 = pattern1.matcher(modification);
-    			Matcher matcher2 = pattern2.matcher(modification);
-    			Matcher matcher3 = pattern3.matcher(modification);
-    			String splitSym = "";
-    			String tempClaStr = "";
-    			String[] splitStr;
-    			if (!matcher1.matches() || !matcher2.matches() || !matcher3.matches()) {
-    				splitSym = (!matcher1.matches()) ? "\\)="
-    					      : ((!matcher2.matches()) ? "\\) =" : "\\)");
-    				splitStr = modification.split(splitSym);
-    				tempClaStr = splitStr[0];
-    				if (splitStr.length > 1) {
-    					this.operator = null;
-    					this.value = splitStr[1];
-    				} else {
-    					this.operator = null;
-    					this.value = null;
-    				}
+    		if (modification.charAt(0) == '(') {    		
+    			List<Integer> equSymbol = new ArrayList<Integer> ();
+        		equSymbol.addAll(isoEqu(modification));
+        		
+        		String temStrL = "";
+        		String temStrR = "";
+        		
+    			if (!equSymbol.isEmpty()) {
+    				temStrL = modification.substring(1, equSymbol.get(0));
+    				temStrR = modification.substring(equSymbol.get(0)+1,modification.length());
+    			} else {
+    				temStrL = modification.substring(1,modification.length()-1);
     			}
-    			ClassMod classMod = new ClassMod();
-    			classMod.classMod(tempClaStr);
-    			this.class_modification = classMod;
+        		if (temStrL.charAt(temStrL.length()-1) == ' ') {
+        			temStrL = temStrL.substring(0, temStrL.length()-1);
+        		} 
```

Comment:
Is there are reason why you are not simply calling the trim() method on the string?